ru/turing/courses/lesson3/Registry.java

```diff
@@ -1,13 +1,85 @@
 package ru.turing.courses.lesson3;
 
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
 
-//todo Задание со звездочкой - сделать Registry через generics.
-// Для этого почитать, что такое дженерики и о том, как заставить класс с ними работать
-// Сделать абстрактный доменный класс (похожий на Animal из прошлого урока для этого)
-public class Registry {
-    private final Map<Object, Object> storage = new HashMap<>(); //todo поменять Object, Object на то, что будет у вашего класса в примере
+import ru.turing.courses.lesson2.animals.Person;
 
-    //todo написать и реализовать методы
+/**
+ * Класс реестра.
+ *
+ * @param <Key>   тип ключа реестра.
+ * @param <Value> тип значения записи реестра.
+ */
+public class Registry<Key extends Number, Value extends Person> {
```

Comment:
```suggestion
public class Registry<Integer, Value extends Person> {
```
Прописываешь для ключа Key extends Number, а потом в мэйне запрашиваешь int. Не легче сразу прописать Integer?)